vyper/semantics/analysis/local.py

```diff
@@ -746,72 +746,38 @@ def _analyse_range_call(node: vy_ast.Call) -> list[VyperType]:
-    :param start: first argument to range()
-    :param end: second argument to range()
-    :param node: range() call node
-    :return: The common types of the arguments
-    """
-    for arg in [start, end]:
-        if not isinstance(arg, vy_ast.Num):
-            error = "Value must be a literal integer, unless a bound is specified"
-            raise StateAccessViolation(error, arg)
-
-    if end.value <= start.value:
-        raise StructureException("End must be greater than start", end)
-    return _get_common_argument_types(node)
-
-
-def _analyse_range_bound(
-    start: vy_ast.VyperNode, end: vy_ast.VyperNode, bound: vy_ast.VyperNode, node: vy_ast.Call
-) -> list[VyperType]:
-    """
-    Check that the arguments to a range(x, bound=N) or range(x, y, bound=N) call are valid.
-    :param start: first argument to range()
-    :param end: second argument to range()
-    :param bound: bound keyword argument to range()
-    :param node: range() call node
-    """
-    if not isinstance(bound, vy_ast.Num):
-        raise StateAccessViolation("Bound must be a literal", bound)
-    if bound.value <= 0:
-        raise StructureException("Bound must be at least 1", bound)
-    if (
-        isinstance(start, vy_ast.Num)
-        and isinstance(end, vy_ast.Num)
-        and end.value - start.value > bound.value
-    ):
-        raise StructureException(
-            f"For loop has invalid number of iterations ({end.value - start.value}),"
-            " the value must be between zero and the bound",
-            node,
-        )
-    return _get_common_argument_types(node)
-
+    all_args = (start, end, *kwargs.values())
+    for arg1 in all_args:
+        validate_expected_type(arg1, IntegerT.any())
 
-def _get_common_argument_types(node: vy_ast.Call) -> list[VyperType]:
-    """
-    Gets the common types of all arguments to a function call.
-    :param node: range() call node
-    :return: list of common types between all arguments
-    """
-    all_args = [*node.args, *[arg.value for arg in node.keywords]]
-    for arg in all_args:
-        validate_expected_type(arg, IntegerT.any())
     type_list = get_common_types(*all_args)
     if not type_list:
         raise TypeMismatch("Iterator values are of different types", node)
+
+    if "bound" in kwargs:
+        bound = kwargs["bound"]
+        if not isinstance(bound, vy_ast.Num):
+            raise StateAccessViolation("Bound must be a literal", bound)
+        if bound.value <= 0:
+            raise StructureException("Bound must be at least 1", bound)
+        if (
+            isinstance(start, vy_ast.Num)
+            and isinstance(end, vy_ast.Num)
+            and end.value - start.value > bound.value
+        ):
+            raise StructureException(
+                f"For loop has invalid number of iterations ({end.value - start.value}), "
+                f"the value must be between zero and the bound",
+                node,
+            )
```

Comment:
i'm wondering if we really want to allow this case (`range(N, M, bound=K)` where `N`, `M` and `K` are all constants). i think maybe we should just reject it?

Reply:
Sure, I can remove it